tests/gateways/test_connection.py

```diff
@@ -108,3 +116,51 @@ def test_s3_server(minio_s3_server):
                 ):
                     # we just want to run make_temp_env and cleanup after
                     pass
+
+
+def test_session_manager_returns_default():
+    """
+    Tests to make sure that our session manager returns a regular
+    CondaSession object when no other session classes are registered.
+    """
+    url = "https://localhost/test"
+    session_obj = session_manager(url)
+
+    assert type(session_obj) == CondaSession
+
+
+@pytest.mark.parametrize(
+    "url, channels, expected",
+    (
+        (
+            "https://repo.anaconda.com/pkgs/main/linux-64/test-package-0.1.0.conda",
+            ("defaults",),
+            "defaults",
+        ),
+        (
+            "https://conda.anaconda.org/conda-forge/linux-64/test-package-0.1.0.tar.bz2",
+            ("conda-forge", "defaults"),
+            "conda-forge",
+        ),
+        (
+            "http://localhost/noarch/test-package-0.1.0.conda",
+            ("defaults", "http://localhost"),
+            "http://localhost",
+        ),
+        ("http://localhost", ("defaults",), None),
+    ),
+)
+def test_get_channel_name_from_url(url, channels, expected):
+    """
+    Makes sure we return the correct value from the ``get_channel_name_from_url`` function.
+    """
+    condarc_data = {"channels": channels}
+    yml = YAML()
+    condarc_stream = StringIO()
+    yml.dump(condarc_data, stream=condarc_stream)
+    condarc_stream.seek(0)
+
+    with temp_context(condarc_stream.read()):
+        channel_name = get_channel_name_from_url(url)
+
+        assert expected == channel_name
```

Comment:
all of these tests can be achieved using the `CONDA_CHANNELS` environment variable so I recommend not introducing `temp_context`:

```suggestion
    monkeypatch.setenv("CONDA_CHANNELS", ",".join(channels))
    reset_context()
    assert context.channels == channels

    assert expected == get_channel_name_from_url(url)
```

Reply:
Thanks. I will check out your suggestion.